gossip/src/main.rs

```diff
@@ -231,21 +236,6 @@ fn process_spy(matches: &ArgMatches, socket_addr_space: SocketAddrSpace) -> std:
     let shred_version = value_t_or_exit!(matches, "shred_version", u16);
     let identity_keypair = keypair_of(matches, "identity");
 
-    let entrypoint_addr = parse_entrypoint(matches);
-
-    let gossip_host = parse_gossip_host(matches, entrypoint_addr);
-
-    let gossip_addr = SocketAddr::new(
-        gossip_host,
-        value_t!(matches, "gossip_port", u16).unwrap_or_else(|_| {
-            solana_net_utils::find_available_port_in_range(
-                IpAddr::V4(Ipv4Addr::UNSPECIFIED),
-                (0, 1),
-            )
-            .expect("unable to find an available gossip port")
-        }),
-    );
```

Comment:
We don't need to change `process_spy` and  `process_rpc_url` signature, right?
We just need to call the added helper function here, instead.